### PR TITLE
Supplement the issue-bot to report on PRs that need attention

### PR DESCRIPTION
**Pull request by mfurtak**

### CODE REVIEW

File: bot.rb [hunk 1]
```diff
@@ -87,7 +162,7 @@ def process_inactive(issue)
           body << "This issue will be auto-closed because there hasn't been any activity for a few months. Feel free to [open a new one](https://github.com/fastlane/fastlane/issues/new) if you still experience this problem 👍"
           client.add_comment(SLUG, issue.number, body.join("\n\n"))
           client.close_issue(SLUG, issue.number)
-          client.add_labels_to_an_issue(SLUG, issue.number, AUTO_CLOSED)
+          client.add_labels_to_an_issue(SLUG, issue.number, [AUTO_CLOSED])
```
Comment (mfurtak) [inline bot.rb]: I'm pretty sure this API requires the value to be an array

File: bot.rb [hunk 2]
```diff
@@ -10,25 +10,43 @@ class Bot
     ISSUE_WARNING = 2
     ISSUE_CLOSED = 0.3 # plus the x months from ISSUE_WARNING
     ISSUE_LOCK = 3 # lock all issues with no activity within the last 3 months
+    UNTOUCHED_PR_DAYS = 14
```
Comment (KrauseFx) [inline bot.rb]: Maybe add a comment next to the variable on what it does 👍

File: bot.rb [hunk 3]
```diff
@@ -49,10 +67,67 @@ def process_closed_issue(issue)
...
+      remove_needs_attention_from(pr) if has_label?(pr, NEEDS_ATTENTION)
+    end
+
+    def notify_action_channel_about(needs_attention_prs)
+      return unless needs_attention_prs.any?
+
+      puts "Notifying the Slack room about PRs that need attention..."
+
+      pr_count = needs_attention_prs.size
+      pr_pluralized = pr_count == 1 ? "PR" : "PRs"
```
Comment (KrauseFx) [inline bot.rb]: Let's hope we need the case where the value is 1 soon 👍

File: bot.rb [hunk 4]
```diff
@@ -49,10 +67,67 @@ def process_closed_issue(issue)
...
+
+      pr_query_link = "<#{NEEDS_ATTENTION_PR_QUERY}|#{pr_count} #{pr_pluralized}>"
+
+      post_body = {
+        text: "#{pr_query_link} have not received any attention in the past #{UNTOUCHED_PR_DAYS} days."
+      }.to_json
+
+      puts post_body
```
Comment (KrauseFx) [inline bot.rb]: Do you want to keep this here?

Comment (mfurtak) [inline bot.rb]: Nope, will remove 👍

File: bot.rb (hunk 3 above)
Comment (mfurtak) [inline bot.rb]: :1st_place_medal:

**Comment by mfurtak** [comment]
My plan is now to introduce a second job scheduler that runs `rake process_all` once per day, so that the #action room isn't getting PR messages every 10 minutes 😅

**Comment by mfurtak** [comment]
I'm a little worried that the two different jobs would run concurrently and get messy when touching the same things, so I've adjusted the code to be able to run on either exclusively issues, or exclusively PRs.

This way, when the daily job and the frequent job eventually run together, they won't actually touch the same things.

**Pull request merged**